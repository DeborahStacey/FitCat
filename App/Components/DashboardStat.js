import React, { Component, PropTypes } from 'react'
import { Text, View } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'

import styles from './Styles/DashboardStatStyles'

class DashboardStat extends Component {
  render () {
    return (
      <View style={styles.statContainer}>
        <View style={styles.statIconView}>
          <Icon name={this.props.icon} style={styles.statIcon} />
        </View>
        <View style={styles.statTextView}>
          <Text style={styles.text}>{this.props.stat} {this.props.unit}</Text>
        </View>
        <View style={styles.statButtonView} >
          <Icon name='caret-right' style={styles.statButton} onPress={this.props.onPress} />
        </View>
      </View>
    )
  }
}

DashboardStat.propTypes = {
  unit: PropTypes.string.isRequired,
  stat: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired
}

export default DashboardStat
