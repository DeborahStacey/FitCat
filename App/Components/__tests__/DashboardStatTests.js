import React from 'react'
import DashboardStat from '../DashboardStat'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const tree = renderer.create(
    <DashboardStat icon='paw' stat='125' unit='steps' onPress={() => {}} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
