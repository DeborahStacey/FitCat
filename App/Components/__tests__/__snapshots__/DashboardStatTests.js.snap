exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "margin": 10,
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 2,
        "paddingRight": 20,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "FontAwesome",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "fontSize": 40,
          },
        ]
      }>
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 4,
        "paddingTop": 10,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={undefined}>
      125
       
      steps
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": undefined,
            "fontFamily": "FontAwesome",
            "fontSize": 12,
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {
            "color": "#D8D8D8",
            "fontSize": 40,
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;
